### PR TITLE
[Helper] Add Accessors for Array types

### PR DESCRIPTION
**Pull request by fredroy**
Array as in fixed_array, not std::array or []

It appeared that it was  not possible to get a R/W Accessor on sofa::type::fixed_array, because it was not defined the alias/type  "T::Size".
After investigation, everything defining operator[] and iterators can use R/W AccessVector.
This is not correct because WriteAccessor defines resize/push_back (dynamic allocation API) and some of this types do not support it (fixed_array, type::Vec, Mat, etc).
So in the current state, you are able to call resize on an accessor of type::Vec, but it fails obviously to compile.

So this PR 
 - introduce the trait `is_fixed_array`, which is true for types defining iterators, operator[] AND the field static_size
 - refine the trait `is_vector`, by adding the test on having defined the function `resize()` (not foolproof but should be sufficient)
 - implement R/W AccessorFixedArray for the container having the trait `is_fixed_array`
- add templated tests for those accessors

A bit breaking because of the change in is_vector, and the fact that RWAccessor<sofa::type::Vec> (and other) will be a RWAccessorFixedArray, not a RWAcessorVector anymore

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: Sofa/framework/Type/src/sofa/type/trait/is_fixed_array.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+******************************************************************************/
+#pragma once
+#include <type_traits>
+
+namespace sofa::type::trait
+{
+
+/// Detect if a type T has iterator/const iterator function, operator[](size_t) and defines a static size
+template<typename T>
+struct is_fixed_array
```
Comment (alxbilger) [inline Sofa/framework/Type/src/sofa/type/trait/is_fixed_array.h:30]: Quite verbose. My version of the trait: https://godbolt.org/z/WTv9aKhvs or a concept in C++20: https://godbolt.org/z/5bj6EcvoG

Comment (damienmarchal) [reply Sofa/framework/Type/src/sofa/type/trait/is_fixed_array.h:30]: c++20 concepts !

Comment (fredroy) [reply Sofa/framework/Type/src/sofa/type/trait/is_fixed_array.h:30]: Calm down guys, I dont have c++20 in this branch 😅

@alxbilger your version does not detect sofa::type::Vec as a type of array. I remind you that it does not inherit type::fixed_array anymore.
And I was merely inspired (not saying copying 😅 ) the trait `is_vector` anyway.

(and yes, sofa::type::Vec is not `is_vector` 🥴)

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Review by bakpaul — APPROVED** [review]

**Pull request merged**